bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/AbstractNetatmoThingHandler.java

```diff
@@ -174,4 +178,32 @@ protected void updateProperties(Integer firmware, String modelId) {
         }
         updateProperties(properties);
     }
+
+    public void updateMeasurements() {
+    }
+
+    public void getMeasurements(NetatmoBridgeHandler handler, String device, @Nullable String module, String scale,
+            List<String> types, List<String> channels, Map<String, Float> channelMeasurements) {
+        if (types.size() != channels.size()) {
+            throw new IllegalArgumentException("types and channels lists are different sizes.");
+        }
+
+        List<Float> measurements = handler.getStationMeasureResponses(device, module, scale, types);
+        if (measurements.size() != types.size()) {
+            throw new IllegalArgumentException("types and measurements lists are different sizes.");
+        }
+
+        int i = 0;
+        for (Float measurement : measurements) {
+            channelMeasurements.put(channels.get(i++), measurement);
+        }
+    }
+
+    public void addMeasurement(Set<String> channelIds, List<String> channels, List<String> types, String channel,
+            String type) {
+        if (channelIds.contains(channel)) {
```

Comment:
Can't you just checked if a channel is linked or not by calling `isLinked(...)`? Then you don't need to track all of the linked channel ids.

Reply:
Ah, nvm, I see you still need it for the method above. In that case is there a way for you to abstract the linked channel id tracking logic into this class so that all the subclasses don't have to implement it?

Reply:
No, the channels that you can get measurements are unique by device and module.